integration_test/KeySequenceJJTest.re

```diff
@@ -2,11 +2,13 @@ open Oni_Model;
 open Oni_IntegrationTestLib;
 
 let keybindings =
-  Some({|
+  Some(
+    {|
 [
-  {"key": "jj", "command": "vim.esc", "when": "insertMode"}
+  {"key": "jj", "command": "oni.vim.esc", "when": "oni.insertMode"}
```

Comment:
Note these changes in particular, which is probably going to break the config of quite a few people. If we want to namespace them, though, we're going to have to at some point anyway, and earlier seems better.

But we should still try to communicate it in some way, so it doesn't come as so much of a surprise. Should we start to consider some kind of in-app change log for breaking changes? Since we still don't do releases we could try to write the version used last to disk, and when it changes show all the breaking changes since that version on the Welcome screen for example.

Reply:
I'd prefer us not to break config / backward compatibility where possible - it's something we need to have a story for anyway, so the sooner we exercise this, the better.

> But we should still try to communicate it in some way, so it doesn't come as so much of a surprise. Should we start to consider some kind of in-app change log for breaking changes?

My preference would be to implement an _upgrader_ - when we load keybindings, to correct the incorrect ones, and send a warning that these are deprecated. We've done this once before (when the schema changed): https://github.com/onivim/oni2/blob/d4a60cf7e4b1260bdcb1f47503440b16d96fb2ff/src/Input/Keybindings.re#L235

Is there a case that conflicts that we couldn't unambiguously upgrade today? Or it would be a mechanical mapping of tokens like:
- `vim.esc` -> `oni.vim.esc`
- `insertMode` -> `oni.insertMode`
etc..

Reply:
That doesn't actually solve the problem though. It just postpones the breakage until later, when it might be less likely to beak people, but at the same time the tolerance for breaks is much lower than in alpha. And all the while we're still just as liable to get conflicts.

In beta and post 1.0 I think backwards-compatibility is the right approach, but in alpha breaking changes should be expected, as long as we communicate it well (which we currently don't).